azure/scope/cluster.go

```diff
@@ -379,12 +379,19 @@ func (s *ClusterScope) SubnetSpecs() []azure.ResourceSpecGetter {
 			Role:              subnet.Role,
 			NatGatewayName:    subnet.NatGateway.Name,
 		}
+		if s.IsVnetManaged() {
+			serviceEndpointList := make([]infrav1.ServiceEndpointSpec, 0, len(subnet.ServiceEndpoints))
```

Comment:
Let's do this `serviceEndpointList` construction in a separate func as it appears to be identical in L405-413 below.

Reply:
done